File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int ap = 0;
        int bp=numbers.length -1;

        while(ap<=bp){
            int sum=numbers[ap]+numbers[bp];
            if(sum>target){
                bp-=1;
            }else if(sum<target){
                ap+=1;
            }else if(sum==target){
                return new int[] {ap+1,bp+1};
            }
        }
        return null;
    }
}

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        vector<int> ans;
        int spt=0;
        int ept=numbers.size()-1;
        while(spt<ept){
            int sum=numbers[spt] + numbers[ept];
            if(sum == target){
                ans.push_back(spt+1);
                ans.push_back(ept+1);
                break;
            }
            else if(sum < target)spt++;
            else ept--;
        }
        return ans;
    }
};